### PR TITLE
[Tensorflow Lite] GPU selection support on Desktop

### PR DESCRIPTION
**Pull request by DwayneDuane**
Adding the ability to select between multiple available GPUs (usually on Desktop environment) for running inference with the OpenCL delegate. By default, if the user does not specify a preference, OpenCL device 0 on platform 0 is still picked by default, so as to not break things on mobile devices where typically only a single OpenCL capable device is available.

### CODE REVIEW

File: tensorflow/lite/CMakeLists.txt [hunk 1]
```diff
@@ -452,7 +452,10 @@ populate_tflite_source_vars("kernels/internal/reference/integer_ops"
 populate_tflite_source_vars("kernels/internal/reference/sparse_ops"
   TFLITE_KERNEL_INTERNAL_REF_SPARSE_OPS_SRCS
 )
-set(TFLITE_PROFILER_SRCS ${TFLITE_SOURCE_DIR}/profiling/platform_profiler.cc)
+set(TFLITE_PROFILER_SRCS 
+    ${TFLITE_SOURCE_DIR}/profiling/platform_profiler.cc
+    ${TFLITE_SOURCE_DIR}/profiling/root_profiler.cc
```
Comment (DwayneDuane) [inline tensorflow/lite/CMakeLists.txt:457]: Without this, linking against libtensorflow-lite.a will result in undefined symbols errors.

**Comment by gbaned** [comment]
Hi @DwayneDuane Can you please resolve conflicts? Thank you!

**Comment by terryheo** [comment]
1. Could you update C++ style? https://www.tensorflow.org/community/contribute/code_style#c_coding_style
2. Please resolve the conflict on tensorflow/lite/CMakeLists.txt

**Comment by DwayneDuane** [comment]
@terryheo @gbaned I just fixed the merge conflict, switched from using `std::pair` to `absl::StatusOr`, and clang formatted the files I touched.

**Comment by gbaned** [comment]
Hi @terryheo Can you please review this PR ? Thank you!

**Comment by gbaned** [comment]
Hi @terryheo Can you please review this PR ? Thank you!

**Comment by impjdi** [comment]
While you can use TFLite on desktop, it's not a supported use case.  We don't want to pollute the mobile inference code with desktop stuff.

**Pull request closed without merging**